Release Notes Window

Jack251970 opened the pull request:
# Add release notes window

- Release notes window will show for one time when Flow is updated.
- In about page, we can open release window manually.
- It will fetch three latest release notes and show them in the window.
- If Flow failed to fetch release notes, it will pop up refresh button.

# Test
![Screenshot 2025-06-07 165139](https://github.com/user-attachments/assets/80695f98-a605-4630-aa0a-23e8931b62b0)
![Screenshot 2025-06-07 165444](https://github.com/user-attachments/assets/4f6e4351-00af-44fb-a485-d8921c5b7c4f)

Jack251970 commented on Flow.Launcher/MainWindow.xaml.cs:
```diff
@@ -132,6 +135,15 @@ private void OnLoaded(object sender, RoutedEventArgs _)
 
+            if (_settings.ReleaseNotesVersion != Constant.Version)
+            {
+                // Update release notes version
+                _settings.ReleaseNotesVersion = Constant.Version;
+
+                var releaseNotesWindow = new ReleaseNotesWindow();
+                releaseNotesWindow.Show();
+            }
+
```
If this is the first launcher, we do not show release notes.

Jack251970 commented:
Force-push for mistaken commits

onesounds commented:
Automatically opening a window after an update can often feel inconvenient for users. This approach is only truly useful when the update is optional and the user is given the choice to proceed. However, since our updates are applied automatically, there's little benefit in showing a separate window.

Instead, I believe the following approach would be more appropriate:

1. In-app popup within the settings page
If we really want to show something, it can appear as a notification or popup when the user opens the settings page.

2. Notification banner at the top of the main page after an update (Recommended)
For example: “The app has been updated. Click here to view the release notes.”
→ Clicking the banner(infobar with button) can either open a webpage or show a popup with the release notes.

3. Add an “Update Info” item at the top of the tray menu after an update
This menu item disappears after being clicked once.

Personally, I think option 2 strikes the best balance. It informs the user without being disruptive.

Additionally, if we’re fetching release notes from GitHub, we should consider that some users might be in regions where GitHub access is restricted. If we only show a "Refresh" button, users may simply respond with “Why isn’t it working?”
→ So we should include a clear message like “Please check your network connection or ensure GitHub is accessible.”

Jack251970 commented:
> Automatically opening a window after an update can often feel inconvenient for users. This approach is only truly useful when the update is optional and the user is given the choice to proceed. However, since our updates are applied automatically, there's little benefit in showing a separate window.
> 
> Instead, I believe the following approach would be more appropriate:
> 
> 1. In-app popup within the settings page
>    If we really want to show something, it can appear as a notification or popup when the user opens the settings page.
> 2. Notification banner at the top of the main page after an update (Recommended)
>    For example: “The app has been updated. Click here to view the release notes.”
>    → Clicking the banner(infobar with button) can either open a webpage or show a popup with the release notes.
> 3. Add an “Update Info” item at the top of the tray menu after an update
>    This menu item disappears after being clicked once.
> 
> Personally, I think option 2 strikes the best balance. It informs the user without being disruptive.
> 
> Additionally, if we’re fetching release notes from GitHub, we should consider that some users might be in regions where GitHub access is restricted. If we only show a "Refresh" button, users may simply respond with “Why isn’t it working?” → So we should include a clear message like “Please check your network connection or ensure GitHub is accessible.”

Added release notes fetch error message as required.

And I implemented a new api method which can show a message box with a button.

Jack251970 commented on Flow.Launcher.Infrastructure/Http/Http.cs:
```diff
@@ -220,5 +220,18 @@ public static async Task<HttpResponseMessage> SendAsync(HttpRequestMessage reque
+                return await client.GetStringAsync(url, token);
+            }
+            catch (System.Exception e)
+            {
```
No need

Jack251970 commented on Flow.Launcher/Notification.cs:
```diff
@@ -67,5 +88,58 @@ private static void LegacyShow(string title, string subTitle, string iconPath)
+                    .AddButton(buttonText, ToastActivationType.Background, guid)
+                    .AddAppLogoOverride(new Uri(Icon))
+                    .Show();
+                _notificationActions.Add(guid, buttonAction);
```
Good!
